examples/vision/swin_transformers.py

```diff
@@ -132,7 +132,7 @@ def __init__(self, drop_prob=None, **kwargs):
     def call(self, x):
         input_shape = tf.shape(x)
         batch_size = input_shape[0]
-        rank = len(input_shape)
+        rank = input_shape.shape.as_list()[0]
```

Comment:
`input_shape.shape.as_list()[0]` seems pretty confusing, don't you think? The shape of a shape?

Just use `len(x.shape)` or `x.shape.rank`.

Reply:
Indeed `len(x.shape)` would achieve the same, but I thought to change it because when you run `model.save`  we get the error:

> TypeError: len is not well defined for symbolic Tensors. (model_2/swin_transformer_4/drop_path_4/Shape:0) Please call `x.shape` rather than `len(x)` for shape information.

Reply:
As I mentioned earlier, we do not use `model.save` in the example, so we could easily skip this if you feel it might get confusing. I just thought to add this in the first place because just in case someone runs this example and then wants to save their model.

Reply:
Both `len(x.shape)` or `x.shape.rank` will work here. Remember, `x.shape` is the static shape and `tf.shape(x)` is the symbolic shape.

Reply:
Thanks, I got a bit confused for a moment there.

Another quick one: Since we just need the shape value might it be easier to just do `rank = x.shape[0]`, which works?

Reply:
`batch_size` is dynamic no?

Reply:
Ah, gotcha. Sorry for the confusion from my end 😅 

Updated as you suggested in the first comment, thanks.